datasets/us/navy/crawler.py

```diff
@@ -0,0 +1,129 @@
+from lxml import html, etree
+import orjson
+from urllib.parse import urljoin
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.zyte_api import fetch_html
+
+BASE_URL = "https://www.navy.mil"
+API_URL = "https://www.navy.mil/API/ArticleCS/Public/GetList?moduleID=709&dpage=%d&TabId=119&language=en-US"
+
+
+def emit_person(context: Context, country: str, source_url: str, role: str, name: str):
+    person = context.make("Person")
+    person.id = context.make_id(country, name, role)
+    person.add("name", name)
+    person.add("position", role)
+    person.add("sourceUrl", source_url)
+
+    position = h.make_position(context, role, country=country, topics=["mil"])
+
+    categorisation = categorise(context, position, is_pep=True)
+    if categorisation.is_pep:
+        occupancy = h.make_occupancy(context, person, position)
+        if occupancy:
+            context.emit(person, target=True)
+            context.emit(position)
+            context.emit(occupancy)
+
+
+def crawl_person(context: Context, item_html: str) -> None:
+    doc = html.fromstring(item_html)
+    link_el = doc.find(".//a")
+    if link_el is None:
+        return
+    url = link_el.get("href")
+    name = link_el.find(".//h2").text_content().strip()
+    role = link_el.find(".//h3").text_content().strip()
+    emit_person(context, "us", url, role, name)
+
+
+def process_page(context: Context, page_number: int):
+    url = API_URL % page_number
+    try:
+        data = context.fetch_json(url)
+    except orjson.JSONDecodeError as e:
+        context.log.error(f"Failed to decode JSON from {url}: {e}")
+        return False, False
+    except Exception as e:
+        context.log.error(f"Failed to fetch JSON from {url}: {e}")
+        return False, False
+
+    if not data:
+        context.log.error(f"No data found for page {page_number}")
+        return False, False
+
+    if isinstance(data, dict):
+        html_data = data.get("data")
+        done = data.get("done", True)
+    else:
+        try:
+            root = etree.fromstring(data)
+            html_data = root.find(".//data").text
+            done = root.find(".//done").text.lower() == "true"
```

Comment:
are....are some pages JSON and some XML and you don't know ahead of time which it's going to be?!

or...is the server responding with XML to your browser, and with JSON to `context.fetch_json` ?